### PR TITLE
[ticket/16425] Add event core.generate_smilies_modify_sql

### PR DESCRIPTION
**Pull request by kasimi**
PHPBB3-16425

Checklist:

- [x] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

https://tracker.phpbb.com/browse/PHPBB3-16425

### CODE REVIEW

File: phpBB/includes/functions_posting.php [hunk 1]
```diff
@@ -109,18 +109,51 @@ function generate_smilies($mode, $forum_id)
...
 	if ($mode == 'window')
 	{
-		$sql = 'SELECT smiley_url, MIN(emotion) as emotion, MIN(code) AS code, smiley_width, smiley_height, MIN(smiley_order) AS min_smiley_order
-			FROM ' . SMILIES_TABLE . '
-			GROUP BY smiley_url, smiley_width, smiley_height
-			ORDER BY min_smiley_order';
+		$sql_ary = [
+			'SELECT'	=> 'smiley_url, MIN(emotion) as emotion, MIN(code) AS code, smiley_width, smiley_height, MIN(smiley_order) AS min_smiley_order',
+			'FROM'		=> [
+				SMILIES_TABLE => 's',
```
Comment (rxu) [inline phpBB/includes/functions_posting.php:115]: I guess the alias should be used with the filed names selected then, i.e. `s.smiley_url` and so on. As well as with the field names in `WHERE` and `ORDER BY` clauses.

File: phpBB/includes/functions_posting.php [hunk 2]
```diff
@@ -109,18 +109,51 @@ function generate_smilies($mode, $forum_id)
...
+			],
+			'GROUP_BY'	=> 'smiley_url, smiley_width, smiley_height',
+			'ORDER_BY'	=> 'min_smiley_order',
+		];
+	}
+	else
+	{
+		$sql_ary = [
+			'SELECT'	=> '*',
```
Comment (rxu) [inline phpBB/includes/functions_posting.php:124]: Same as above, and with the filed names below as well.

**Comment by dark-1** [comment]
Can somebody restart : https://ci.appveyor.com/project/phpBB/phpbb/builds/32071802/job/lsi3a964dw58g35r
Also what happened to Travis ???

File: phpBB/includes/functions_posting.php [hunk 3]
```diff
@@ -109,18 +109,51 @@ function generate_smilies($mode, $forum_id)
...
+	}
+
+	/**
+	* Modify the SQL query that fetches the smilies
+	*
+	* @event core.generate_smilies_modify_sql
+	* @var string	mode		Smiley mode, either window or inline
+	* @var int		forum_id	Forum where smilies are generated, or 0 if composing a private message
+	* @var array	sql_ary		Array with SQL query data
+	* @since 3.2.10-RC1
```
Comment (marc1706) [inline phpBB/includes/functions_posting.php:140]: Can you add the `@since 3.3.1-RC1` as well? Thanks.

**Pull request merged**